integration-tests/src/checks/correct_filtering_different_topics.rs

```diff
@@ -0,0 +1,66 @@
+use crate::setup::constants::{
+    MOCK_SUBGRAPH_GOERLI, MOCK_SUBGRAPH_GOERLI_2, MOCK_SUBGRAPH_MAINNET,
+};
+use crate::setup::test_radio::run_test_radio;
+use crate::utils::RadioRuntimeConfig;
+use colored::Colorize;
+use poi_radio::{LocalAttestationsMap, MessagesArc, RemoteAttestationsMap};
+
+use tracing::{debug, info};
+
+fn post_comparison_handler(_messages: MessagesArc, _block: u64, _subgraph: &str, _prev_len: usize) {
+}
+
+fn test_attestation_handler(
+    _block: u64,
+    _remote: &RemoteAttestationsMap,
+    _local: &LocalAttestationsMap,
+) {
+}
+
+fn success_handler(messages: MessagesArc) {
+    let messages = messages.lock().unwrap();
+
+    // Maybe pass in dynamic count here too
+    if messages.len() >= 5 {
+        debug!("messages {:?}", messages);
+
+        info!("5 or more valid messages received! Checking content topics");
+        assert!(
+            messages
+                .iter()
+                .any(|m| m.identifier == MOCK_SUBGRAPH_MAINNET),
+            "No message found with topic {}",
+            MOCK_SUBGRAPH_MAINNET
+        );
+        assert!(
```

Comment:
Trying to think if this should be a bit more strict, maybe we could switch to if there are no messages other than the configured subgraphs, instead of only checking for MOCK_SUBGRAPH_GOERLI?

Reply:
✅